schemas/network.yml

```diff
@@ -16,11 +16,20 @@
           type: keyword
           description: >
             Name given by operators to sections of their network.
-
-    - name: protocol
+    - name: network
```

Comment:
This would be `network.network`. Not sure if that make sense. Any alternatives we could use?

Reply:
Yeah, I'm not a huge fan, but it aligns with the OSI model which is a standard with regard to naming these things. Does anyone else have any thoughts? @webmat @MikePaquette ?

Reply:
I think I'd rather stick with `network.internet`, which is also weird, but at least is not a repetition like `network.network`...

Or perhaps `network.layer3`? Not a super fan of that either because it's inconsistent vs using the actual names for all the others. But I wanted to mention that option.

Reply:
Why not to use network.protocol.* naming which is not confusing?

Reply:
Putting OSI aside: What will the common user look for naming wise? What about the suggestion from @vbohata 

What is still not fully clear to me why we can't mix them all in one field. Are the aggregations we can't do then?

One more option: What about `network.layer: layer_name` together with `network.protocol`?

Reply:
Putting all into one field can not work as expected. Some logs contains for example both network.protocol.application and network.protocol.transport and users want to do term aggs per application protocol, or ...

Reply:
I agree with @vbohata, people will want to aggregate on these. You may want to see ratio of UDP vs TCP, but if you then have HTTP, SMTP, etc intermingled, that doesn't help. Same thing for aggregating around the application level protocol, you don't want to see TCP and UDP pop up there.

Let's make @vbohata's proposal explicit, to see it in action and make sure we all understand the same thing:

```
network.protocol.link: arp, mac...
network.protocol.network: ip, ipx, icmp
network.protocol.transport: udp, tcp...
network.protocol.application: http, ftp, dns...
```

@vbohata is that what you had in mind?

I do think it helps reduce the weirdness of seeing `network.network`, that's for sure :-)

Another thing to consider is that the top two layers in the list above will be used less often than the bottom two, in many cases. So a bit of naming weirdness there may not be the end of the world. I would expect to see `link` and `network` in some of the networking logs for people managing data centers. I would expect to see it less often in cloud environments, where usually the networking layer is managed by the provider / taken for granted.

Also, these fields being in ECS doesn't mean all logs using ECS must fill them, IMO. I don't think web server logs should include the `link` and `network` layer, as their value would absolutely always be the same. I could hardly imagine a situation where I need to compare web server logs vs BGP logs directly (as in: aggregation to see which one you have more of). So the logs centered around managing the network will have the lower level ones, and the logs centered around the applications will have only `application` (e.g. web logs) or `application` and `transport` (e.g. dns, IDS).

Reply:
Couple thoughts:
a specific section for wireless info might be nice, unless that fits into another bucket (e.g. ssid # / name, not sure how likely it would be to be able to tell physical network info from anything besides e.g. wireless alerts, but having a way to normalize e.g. WIPS alert info / RADIUS WPA2-ENT would be nice)

Something like this might eliminate some of the understanding issues:
network.protocol  (IP / GRE / EIGRP, etc)
network.transport (tcp/udp)
network.application (http/telnet/ssh/smtp/tls)
network.application.id (L7 id - twitter, facebook, etc)

but adding in e.g. COS / DSCP, MPLS tags, etc and some of the other items might be useful from an APM standpoint . IPFIX / Netflow9 would probably be a decent place to start adding relevant network data.

https://www.iana.org/assignments/ipfix/ipfix.xhtml
some of the fields may well be superfluous (delta counts, aggregate counts, etc)

and layer 3 is definitely the NETWORK layer :)

Reply:
and 1 question - as I look thru the various models are the ymls meant to support a given type of logging source (e.g. network = netflow / network sensor / basic fw network telemetry?)  If so it would be good to discretely identify the expected sources & target info (e.g. NPM people want QOS/DSCP for monitoring, APM people should too... but a normalized netflow target is likely to look significantly different than a normalized FW flow/connection target data type)

Reply:
Hi @dainperkins! ECS is meant to document a common event schema that would help make different event/log sources look more similar to one another. This will make it easier to find the basics in any event stream. The use cases we are mostly focusing on at the moment is operational monitoring as well as security.

By its nature, though, ECS will never encompass all use cases for all sources. Any given stream is likely to have fields that don't fit in ECS. You can use your custom fields around the documented ECS fields.

If you see existing fields that aren't quite defined correctly to fit some needs, or if you see generic fields that are missing, it's totally fair to open the discussion around it, though. So feel free to open issues or PRs with specific suggestions.

Reply:
from an NSM perspective where we might have knowledge of more than one layer of protocol in a given event, this is fantastic:

> network.protocol (IP / GRE / EIGRP, etc)
> network.transport (tcp/udp)
> network.application (http/telnet/ssh/smtp/tls)
> network.application.id (L7 id - twitter, facebook, etc)

Namely, HTTP is absolutely an OSI layer 7 protocol, but it adds value if we have signatures or behavior models that are picking up the next layer application model of twitter, facebook, etc.

Reply:
Thanks Derek!

Note that we can't have both `network.application` and `network.application.id` at the same time in the field mapping. The following would work, though: `network.application` and `network.application_id`.

I like @dainperkins's suggestion of adding `network.application_id`, but for now I think we should focus on finding sensible names for each network layer. We can add `_id` later. I'll try to close on this sooner than later. Based on the conversations here I would be inclined to go with:

* `network.protocol`: e.g. IP, GRE, EIGRP
* `network.transport`: TCP, UDP
* `network.application`: HTTP, HTTPS, SSH, TLS, SMTP

Let me know if there's any big disagreement on this.

Reply:
Pro: shorter, con: not self describing enough
Example: User wants to filter all events related to program "named" and "UDP" protocol. Until deeper look at logged data it seems network.application will be "named" and network.protocol will be "udp". In longer variant everyone immediately see "network.protocol.transport" is the correct field and network.protocol.application will probably not contain program name ...

Reply:
My concern is that `network.protocol` in network sensor applications (bro, suricata, etc) is almost always the transport layer protocol (in the OSI sense). ICMP, GRE, and EIGRP, and of course TCP and UDP are all IP-encapsulated protocols, in that they all require a valid IP header first. Arp is probably one of the common protocols that you might log that sits at the network layer. 

All that said, there are certainly times where one wouldn't be using IP as the network protocol, but in general it's at least a 95% solution. I'm not inclined to accrue the cost of adding a field to simply mark all my log entries as `IP`. And after I do that, I have to rename all the `network.protocol` fields to `network.protocol.transport`. 

I will adopt the `.network.application` and `network.application_id` annotation, however. Thanks for that suggestion @webmat